### PR TITLE
Avoid leaks in Long GC disruption tests

### PR DESCRIPTION
**Pull request by jasontedor**
We can leak disrupted threads here since we never wait for them to complete after freeing them from their loops. This commit addresses this by joining on disrupted threads, and addresses fallout from trying to join here.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/disruption/LongGCDisruption.java [hunk 1]
```diff
@@ -228,23 +237,31 @@ protected boolean stopNodeThreads(Set<Thread> nodeThreads) {
                 if (thread.isAlive() && nodeThreads.add(thread)) {
                     liveThreadsFound = true;
                     logger.trace("stopping thread [{}]", threadName);
-                    thread.suspend();
-                    // double check the thread is not in a shared resource like logging. If so, let it go and come back..
                     boolean safe = true;
```
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/test/disruption/LongGCDisruption.java]: I wonder if it'd be save is we defaulted to safe = false and only set it to true at the bottom of the try block. That way if we throw anything in the try block we'd resume the thread.

**Review by nik9000 — APPROVED** [review]
Just to put this in my own words: we had a problem where we'd suspend threads in a critical section and not resume them because we tried to log first and that would blow up with and InterruptedException. So moving the log message below the resume fixes this.

We didn't notice this issue because we weren't joining on the disrupting thread. Not joining also causes trouble because we could start the blockDetectingThread while we have a thread in a critical section suspended. Even if we fixed the log change so that we eventually resumed it, we'd still get weird results because it was detected as blocked and it was in a critical section.

File: test/framework/src/main/java/org/elasticsearch/test/disruption/LongGCDisruption.java (hunk 1 above)
Comment (ywelsch) [inline test/framework/src/main/java/org/elasticsearch/test/disruption/LongGCDisruption.java]: ++

**Review by ywelsch — APPROVED** [review]
LGTM

File: test/framework/src/main/java/org/elasticsearch/test/disruption/LongGCDisruption.java (hunk 1 above)
Comment (jasontedor) [inline test/framework/src/main/java/org/elasticsearch/test/disruption/LongGCDisruption.java]: I pushed 50e2058b4eccdb13925a6d7fac9596dc4c1dd311.

**Pull request merged**